src/components/tree/phyloTree/types.ts

```diff
@@ -0,0 +1,260 @@
+  cx?: number
+  cy?: number
+  d?: any
+  r?: number
+  "stroke-width"?: number
+  visibility?: Visibility
+
+  // TODO: This should be `string | number`, conditional on layout
+  x: any
+
+  // TODO: This should be `string | number`, conditional on layout
+  y: any
+}
+
+// Apply fixes from above
+export type SVGProperty = Extract<keyof CSSStyleDeclarationWithHyphens | keyof CustomCSSStyleDeclaration, string>
+
+export interface PhyloNode extends Partial<Omit<CSSStyleDeclarationWithHyphens, keyof CustomCSSStyleDeclaration>>, CustomCSSStyleDeclaration {
+  angle?: number
+  branch?: [string, string]
+  branchStroke?: string
+  conf?: [number, number]
+
+  /** SVG path */
+  confLine?: string
+
+  crossDepth?: number
+  depth?: number
+  displayOrder?: number
+  displayOrderRange?: number[]
+  fill?: string
+  inView: boolean
+  n: ReduxNode
+  pDepth?: number
+  px?: number
+  py?: number
+  rot?: number
+  smallBigArc?: boolean
+  tau?: number
+  that: PhyloTree
+  tipStroke?: string
+  update?: boolean
+
+  /** SVG path */
+  vaccineCross?: string
+
+  w?: number
+  xBase?: number
+  xCBarEnd?: number
+  xCBarStart?: number
+  xCross?: number
+  xTip?: number
+  yBase?: number
+  yCBarEnd?: number
+  yCBarStart?: number
+  yCross?: number
+  yTip?: number
+}
+
+export interface ScatterVariables {
+  showBranches?: boolean
+  showRegression?: boolean
+  x?: string
+  xContinuous?: boolean
+  xDomain?: number[]
+  xTemporal?: boolean
+  y?: string
+  yContinuous?: boolean
+  yDomain?: number[]
+  yTemporal?: boolean
+}
+
+export interface Params {
+  branchLabelFill: string
+  branchLabelFont: string
+  branchLabelFontWeight: number
+  branchLabelKey: string | false
+  branchLabelPadX: number
+  branchLabelPadY: number
+  branchStroke: string
+  branchStrokeWidth: number
+  confidence?: boolean
+  fillSelected: string
+  fontFamily: string
+  grid?: boolean
+  majorGridStroke: string
+  majorGridWidth: number
+  mapToScreenDebounceTime: number
+  minorGridStroke: string
+  minorGridWidth: number
+  minorTicks: number
+  orientation: [number, number]
+  radiusSelected: number
+  regressionStroke: string
+  regressionWidth: number
+  showAllBranchLabels?: boolean
+  showGrid: boolean
+  showTipLabels?: boolean
+  tickLabelFill: string
+  tickLabelSize: number
+  tipFill: string
+  tipLabelBreakL1: number
+  tipLabelBreakL2: number
+  tipLabelBreakL3: number
+  tipLabelFill: string
+  tipLabelFont: string
+  tipLabelFontSizeL1: number
+  tipLabelFontSizeL2: number
+  tipLabelFontSizeL3: number
+  tipLabelPadX: number
+  tipLabelPadY: number
+  tipLabels: boolean
+  tipRadius: number
+  tipStroke: string
+  tipStrokeWidth: number
+}
+
+export interface Callbacks {
+  onTipHover: (d: PhyloNode) => void
+  onTipLeave: (d: PhyloNode) => void
+  onTipClick: (d: PhyloNode) => void
+  onBranchHover: (d: PhyloNode) => void
+  onBranchLeave: (d: PhyloNode) => void
+  onBranchClick: (d: PhyloNode) => void
+  tipLabel: (d: PhyloNode) => void
+}
```

Comment:
i would be tempted to define the function sig once and then use that for the callback values:

```
interface CallbackMethodSig {(d:PhyloNode)=> void }
export interface Callbacks {
  onTipHover: CallbackMethodSig
  onTipLeave: CallbackMethodSig
...
```

Reply:
Done: https://github.com/nextstrain/auspice/blob/2af5974e4631a6768727b2a21bae1182bb354047/src/components/tree/phyloTree/types.ts#L347-L357